src/ott/datasets.py

```diff
@@ -152,3 +319,52 @@ def create_gaussian_mixture_samplers(
   )
   dim_data = 2
   return train_dataset, valid_dataset, dim_data
+
+
+def create_sklearn_samplers(
+    source_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    target_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    train_batch_size: int = 256,
+    valid_batch_size: int = 256,
+    rng: Optional[jax.Array] = None,
+) -> Tuple[Dataset, Dataset, int]:
+  """Sklearn samplers for :class:`~ott.solvers.nn.neuraldual.W2NeuralDual`.
+
+  Args:
+  source_kwargs: kwargs to initialize source sampler
+  target_kwargs: kwargs to initialize source sampler
+  train_batch_size: the training batch size
+  valid_batch_size: the validation batch size
+  rng: initial PRNG key
+
+  Returns:
+  The dataset and dimension of the data.
+  """
+  rng = jax.random.PRNGKey(0) if rng is None else rng
+  rng1, rng2, rng3, rng4 = jax.random.split(rng, 4)
+  train_dataset = Dataset(
+      source_iter=iter(
+          SklearnDistribution(
+              init_rng=rng1, batch_size=train_batch_size, **source_kwargs
+          )
+      ),
+      target_iter=iter(
+          SklearnDistribution(
+              init_rng=rng2, batch_size=train_batch_size, **target_kwargs
+          )
+      ),
+  )
+  valid_dataset = Dataset(
+      source_iter=iter(
+          SklearnDistribution(
+              init_rng=rng3, batch_size=valid_batch_size, **source_kwargs
+          )
+      ),
+      target_iter=iter(
+          SklearnDistribution(
+              init_rng=rng4, batch_size=valid_batch_size, **target_kwargs
+          )
+      ),
+  )
+  dim_data = 2
```

Comment:
Why is this hardcoded here? I would consider removing this and just returning the 2 datasets.